systemd/systemd.go

```diff
@@ -1416,14 +1419,16 @@ X-SnapdOrigin={{.}}
 {{- end}}
 `
 
-// We want kernel-modules components to be mounted before modules are
+// We want some snaps and components to be mounted before modules are
 // loaded (that is before systemd-{udevd,modules-load}).
 const beforeDriversLoadUnitTmpl = `[Unit]
 Description={{.Description}}
 DefaultDependencies=no
 After=systemd-remount-fs.service
 Before=sysinit.target
 Before=systemd-udevd.service systemd-modules-load.service
+Before=snapd.mounts.target
+Before=local-fs.target
```

Comment:
I am trying to figure out now, why we cannot use the same template, and only add `Before=systemd-udevd.service systemd-modules-load.service` only for kernel module snaps.  I do not see the dependency that needs to be removed.

Reply:
There is an `After=snapd.mounts-pre.target` in the regular mount units that probably we don't want here. Anyway, even if that was not the case and we could have very similar units by using `DefaultDependencies=no` and adding manually the dependencies I prefer to have separate templates precisely so regular units do not need to be explicit about dependencies and they look simpler.

As a side note, I've read again systemd.mount(5) and I have changed:
```
After=systemd-remount-fs.service
```
to
```
After=local-fs-pre.target
```
as that seems to be the default dependency (should not make a big difference).

Reply:
My point is that I do not think that `DefaultDependencies=no` is needed. I do not see what introduced dependencies are breaking it.

Could we know why `After=snapd.mounts-pre.target` is bad for this one? I do not think it has to be after module loading.

Reply:
Ok, I got you know. Yeah, definitely in the end I am repeating the default dependencies in the early mount units. I have changed things so we use only one template. Re: `After=snapd.mounts-pre.target` I think we can have it also for early mounts. It has an after dependency on `zfs-mount.service` due to https://bugs.launchpad.net/ubuntu/+source/snapd/+bug/1922293 , which should be fine.